src/transformers/generation/utils.py

```diff
@@ -820,11 +820,20 @@ def _get_logits_warper(
         # instantiate warpers list
         warpers = LogitsProcessorList()
 
+        # In beam methods, we need to keep at least one non-eos token to explore continuations that might have a
+        # better score (i.e. keep len(generation_config.eos_token_id) + 1)
```

Comment:
it's not evidend that generation_config is a list as it usually is an int no? 
```suggestion
        # In beam methods, we need to keep at least one non-eos token to explore continuations that might have a
        # better score (i.e. keep len(list(generation_config.eos_token_id)) + 1)
```